### ISSUE TITLE
Add shuffle for pytorch dataloader

### ISSUE THREAD

**Issue by DeliangFan** [uber/petastorm#341]
Since the shuffling is not well supported by petastorm pytorch dataloader.  We need to improve the shuffle feature.

**Comment by selitvin** [comment]
Solved by #382
